test/mlir-cpu-runner/mlir_runner_utils.cpp

```diff
@@ -175,3 +175,29 @@ extern "C" void
 print_memref_vector_4x4xf32(StridedMemRefType<Vector2D<4, 4, float>, 2> *M) {
   printMemRef(*M);
 }
+
+extern "C" void print_memref_anyrank_f32(UnrankedMemRefType *M) {
+  printUnrankedMemRefMetaData(std::cout, *M);
+  int rank = M->rank;
+  void *ptr = M->descriptor;
+
+  switch (rank) {
+  case 0:
+    print_memref_0d_f32(static_cast<StridedMemRefType<float, 0> *>(ptr));
+    break;
+  case 1:
+    print_memref_1d_f32(static_cast<StridedMemRefType<float, 1> *>(ptr));
```

Comment:
Can we avoid the extra level of indirection and just use the underlying?
`template <typename T, int N> void printMemRef(StridedMemRefType<T, N> &M)`

Then we can just remove the individual functions and have a single unranked print API to hook into